tests/TestCase/Database/Type/DecimalTypeTest.php

```diff
@@ -177,6 +194,12 @@ public function testMarshal()
 
         $result = $this->type->marshal(['3', '4']);
         $this->assertNull($result);
+
+        // This test is to indicate the problem that will occur if you use
+        // float/double values which get converted to scientific notation by PHP.
+        // To avoid this problem always using strings to indicate decimals values.
+        $result = $this->type->marshal(1234567890123456789.2);
```

Comment:
Marshalling should generally get strings, so this situation should be rare. Should there be a test added for values with 10+ of decimal places not losing precision?

Reply:
I have added more tests.